README.md

```diff
@@ -1,32 +1,120 @@
+
+# Lasso's Value Proposition
+### Lasso Controller
+Lasso standardizes what a controller is with its controller package. Lasso introduces factories for managing caches, controllers, and clients within a project. This is imperative for rancher which manages many instances of many types.
+
+The lasso controller uses the client-go work-queue. Lasso registers an event handler that processes the work-queue. The work-queue oriented handler makes it so only one event handler needs to be added for each Kubernetes type. Now registering a handler just adds it to the type's controller's list of handlers. Enqueueing an objects adds it to the work-queue. Each object in the work-queue will then be processed by workers running in go routines. The controller's event handler will re-enqueue any object that causes one or more of the controller's handlers to return an error that is not of type `ErrIgnore`. This work-queue can also have objects enqueued from other sources. This functionality is exposed through the controller's Enqueue function. This transforms controllers from only being reactive to Kubernetes resources and cache re-syncs, to also being invokable by code.
+
+### Lasso Shared Controller
+The sharedController type embeds the Controller interface to expose it's controller's functionality. The sharedController also exposes the RegisterHandler function along with the Client function. RegisterHandler adds the passed handler to the controller's handler list and client returns a client for the sharedController corresponding resource type; sharedController has a GVK field (Group Version Kind) which indicates what type it is for.
+
+The sharedController can create its underlying controller. It will only do so once a handler is registered to the sharedController.
+
+### Lasso Client
+Wraps Kubernetes client. Contains GVK fields for managing the client in a parent struct such as the sharedClientyFactory.
+
+### Lasso Cache
+The lasso Cache is a light wrapper around around a [SharedIndexInformer from client-go](https://pkg.go.dev/k8s.io/client-go/tools/cache#SharedIndexInformer).
+
+### Lasso Shared Cache Factory
+The shared cache factory stores the Caches and allows them to be managed via GVK structs. Once the Caches have been created once they can be reused.
+
+### Lasso Shared Client Factory
+The Shared Client Factory exists so that after a client for a GVK has been created once it can be reused.
+
+### Lasso Shared Controller Factory
+The SharedControllerFactory ties everything above together. It allows for a single entry point to access clients, caches, and handlers- all of which it manages for you. It is possible to mainly interact with the shared controller factory. It will create all the underlying controllers, clients, and caches you need. It will ensure there are not duplicates of these objects. It will prevent race conditions.
```

Comment:
> It will ensure there are not duplicates of these objects

It might be helpful to show an example of what this means, like if I call `controllerFactory.ForObject` twice I won't get a duplicate controller?

Reply:
I was trying to illustrate these relationships with this [diagram](https://github.com/rancher/lasso/pull/38/files#diff-37acf97508e174947934159a6676cf2cdc0200fce039ea6bda1325e710aa8e1a) below. I thought it would be sufficient between this paragraph and the diagram, but is there something specific that is being lost?

Reply:
Yes that makes it much clearer, sorry for missing that